src/main/java/apoc/index/IndexUpdateTransactionEventHandler.java

```diff
@@ -0,0 +1,114 @@
+package apoc.index;
+
+import org.neo4j.graphdb.GraphDatabaseService;
+import org.neo4j.graphdb.Node;
+import org.neo4j.graphdb.event.TransactionData;
+import org.neo4j.graphdb.event.TransactionEventHandler;
+import org.neo4j.graphdb.index.Index;
+import org.neo4j.graphdb.index.IndexManager;
+import org.neo4j.kernel.internal.GraphDatabaseAPI;
+import org.neo4j.logging.Log;
+
+import java.util.*;
+
+/**
+ * a transaction event handler that updates manual indexes based on configuration in graph properties
+ * @author Stefan Armbruster
+ */
+public class IndexUpdateTransactionEventHandler extends TransactionEventHandler.Adapter<Void> {
+
+    private final GraphDatabaseService graphDatabaseService;
+    private final Log log;
+
+    /** maintain a transposed version of configuration of all legacy indexes */
+    private Map<String, Map<String, Index<Node>>> propertyNameAndLabelMatrix = null;
```

Comment:
Perhaps it would be easier to have an `Label.property` key? and a simpler map structure?